src/library/Hooks/useTimeLeft/index.tsx

```diff
@@ -117,58 +114,63 @@ export const useTimeLeft = ({
     ReturnType<typeof setInterval> | undefined
   >(undefined);
   const secIntervalRef = useRef(secInterval);
-
-  // refresh callback counter
-  let r = refreshInterval;
-
-  // refresh effects.
-  useEffect(() => {
-    // handler for handling timeleft refresh.
-    //
-    // either handle regular timeleft update, or refresh `to` via `refreshCallback` every
-    // `refreshInterval` seconds.
-    const handleRefresh = () => {
-      if (r !== 0) {
-        setTimeleft(getTimeleft());
-      } else {
-        if (refreshCallback) {
-          setStateWithRef(fromNow(refreshCallback()), setTo, toRef);
+  if (TimeleftHook) {
```

Comment:
useEffects should **Never** be scoped or wrapped in anything. They need to be top level